### PR TITLE
Removes bind_util

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/11471

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
- make sure that you can claim promotion
- make sure that you can do contribution (one time tip, recurring tip and ac)

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_info.cc [hunk 1]
```diff
@@ -424,15 +421,10 @@ void DatabaseContributionInfo::OnGetContributionReport(
...
+    std::shared_ptr<ledger::ContributionInfoList> shared_contributions,
     ledger::GetContributionReportCallback callback) {
-  ledger::ContributionInfoList contribution_list;
-  braveledger_bind_util::FromStringToContributionList(
-      contribution_list_string,
-      &contribution_list);
-
   ledger::ContributionReportInfoList report_list;
-  for (auto& contribution : contribution_list) {
+  for (auto& contribution : *shared_contributions) {
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_info.cc:427]: Can this be const?

File: vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_info.cc [hunk 2]
```diff
@@ -543,14 +536,9 @@ void DatabaseContributionInfo::OnGetList(
...
-    const std::string& contribution_list_string,
+    std::shared_ptr<ledger::ContributionInfoList> shared_contributions,
     ledger::ContributionInfoListCallback callback) {
-  ledger::ContributionInfoList contribution_list;
-  braveledger_bind_util::FromStringToContributionList(
-      contribution_list_string,
-      &contribution_list);
-
-  for (auto& contribution : contribution_list) {
+  for (auto& contribution : *shared_contributions) {
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_info.cc:541]: Can this be const?

File: vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue.cc [hunk 3]
```diff
@@ -87,20 +89,18 @@ void DatabaseContributionQueue::OnInsertOrUpdate(
...
-      braveledger_bind_util::FromStringToContributionQueue(queue_string);
-
-  if (!queue) {
+  if (!shared_queue) {
     BLOG(0, "Queue is null");
     callback(ledger::Result::LEDGER_ERROR);
     return;
   }
 
+  auto queue = std::move(*shared_queue);
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue.cc:98]: Is this necessary?

File: vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion.cc [hunk 4]
```diff
@@ -392,38 +388,36 @@ void Promotion::OnCompletedAttestation(
...
-      braveledger_bind_util::FromStringToPromotion(promotion_string);
-
-  if (!promotion_ptr) {
+  if (!shared_promotion) {
     BLOG(1, "Promotion is null");
     callback(ledger::Result::LEDGER_ERROR, nullptr);
     return;
   }
 
+  auto promotion = std::move(*shared_promotion);
```
Comment (emerick) [inline vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion.cc:412]: Is this necessary?

File: vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue.cc (hunk 3 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue.cc:98]: sadly without this it's not working, do you maybe have a way how to remove this? compiler was complaining a lot :)

Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue.cc:98]: found a way 😄 fixing now

Comment (emerick) [reply vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue.cc:98]: It seemed like you could just do this, but maybe I'm wrong:

```
  publishers_->InsertOrUpdate(
      shared_queue->id,
      std::move(shared_queue->publishers),
      callback);
```

Comment (emerick) [reply vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue.cc:98]: Awesome!

File: vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution.cc [hunk 5]
```diff
@@ -437,42 +431,41 @@ void Contribution::OnEntrySaved(
...
 
-  if (queue->amount > 0) {
+  if ((*shared_queue)->amount > 0) {
     auto save_callback = std::bind(&Contribution::OnQueueSaved,
       this,
       _1,
       wallet_type,
       balance,
-      braveledger_bind_util::FromContributionQueueToString(queue->Clone()));
+      std::make_shared<ledger::ContributionQueuePtr>((*shared_queue)->Clone()));
```
Comment (zenparsing) [inline vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution.cc:440]: Can the shared pointer just be passed in to bind directly, without requiring the clone?

File: vendor/bat-native-ledger/src/bat/ledger/internal/report/report.cc [hunk 6]
```diff
@@ -74,27 +69,20 @@ void Report::OnTransactions(
...
-  const std::string callback_monthly_string =
-      braveledger_bind_util::FromMonthlyReportToString(
-          std::move(monthly_report));
+  (*shared_report)->transactions = std::move(transaction_report);
 
   auto contribution_callback = std::bind(&Report::OnContributions,
       this,
       _1,
-      callback_monthly_string,
+      std::make_shared<ledger::MonthlyReportInfoPtr>(std::move(*shared_report)),
```
Comment (zenparsing) [inline vendor/bat-native-ledger/src/bat/ledger/internal/report/report.cc:85]: Can the shared pointer be passed directly into bind?

File: vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution.cc (hunk 5 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution.cc:440]: fixed

File: vendor/bat-native-ledger/src/bat/ledger/internal/report/report.cc (hunk 6 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/report/report.cc:85]: fixed

**Pull request merged**